File: app/src/main/java/in/example/rahul/vegcartpro/utils/SharedPreferenceUtils.java
package in.example.rahul.vegcartpro.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPreferenceUtils {

    Context mContext;
    SharedPreferences pref;

    public SharedPreferenceUtils(Context context){
        this.mContext= context;
        try {
            pref= this.mContext.getSharedPreferences("Preference", Context.MODE_PRIVATE);
        } catch (Exception e){
            e.printStackTrace();
        }
    }
    public void setFlowerVeg(String veg){
        SharedPreferences.Editor editor= pref.edit();
        editor.putString("id", veg);
        editor.commit();
    }
    public String getFlowerVeg(){
       if (pref.contains("id")) return pref.getString("id", "");
       else return "null";
    }
}

File: app/src/main/java/in/rahulkhatri/vegcartpro/utils/SharedPreferenceUtils.kt
package `in`.rahulkhatri.vegcartpro.utils

import android.content.Context
import android.content.SharedPreferences

class SharedPreferenceUtils(var mContext: Context?) {
//    var mContext: Context? = null
    var pref: SharedPreferences? = null
init {
//    mContext = context
    try {
        pref = mContext?.getSharedPreferences("Preference", Context.MODE_PRIVATE)
    } catch (e: Exception) {
        e.printStackTrace()
    }
}
   /* fun SharedPreferenceUtils(context: Context?) {
        mContext = context
        try {
            pref = mContext!!.getSharedPreferences("Preference", Context.MODE_PRIVATE)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }*/

    fun setFlowerVeg(veg: String?) {
        val editor = pref?.edit()
        editor?.putString("id", veg)
        editor?.apply()
    }

    fun getFlowerVeg(): String? {
        return if (pref!!.contains("id")) pref?.getString("id", "") else "null"
    }

    fun setCategoryItem(veg: String?) {
        val editor = pref?.edit()
        editor?.putString("id", veg)
        editor?.apply()
    }

    fun getCategoryItem(): String? {
        return if (pref!!.contains("id")) pref?.getString("id", "") else "null"
    }
}